autogluon/utils/tabular/ml/models/tab_transformer/utils.py

```diff
@@ -0,0 +1,186 @@
+from tqdm import tqdm
+from torch.utils.data import Dataset, DataLoader
```

Comment:
All torch imports need to be moved to the exact place where they are actually needed and then preceded by `try_import_torch`.  This way, the user can use AutoGluon even without torch installed, and will only experience a failure when they try to use a torch-dependent function (ie. try to fit() with TabTransformer).

Reply:
For files which are only imported by other torch dependent files, no need to do the try_import_torch in them, just do it in the outer files by doing try_import_torch before import the file (and making the file import lazy). This will simplify the effort.